src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -4419,6 +4441,8 @@ else if (serverMajorVersion >= 9) // Yukon (9.0) --> TDS 7.2 // Prelogin disconn
 
         len2 = len2 + 1; // add 1 to length becaue of FeatureEx terminator
 
+        len2 = len2 + writeUTF8SupportFeatureRequest(false, tdsWriter);
```

Comment:
This is a little convoluted, the function basically always returns "5". It also has a Boolean to turn on/off the actual writing. Shouldn't we remove the length calculation from this function and just have a static variable which equals 5? Can also remove the Boolean parameter.

Reply:
This is done to keep the same structure for all feature extension requests, in case we calculate the length differently in the future. Take a look at `writeFedAuthFeatureRequest`, `writeAEFeatureRequest`, `writeDataClassificationFeatureRequest`, The methods are implemented the same way in .NET driver too.

Reply:
It makes sense with fedAuth because the length is actually variable. For the others, it seems there's a set value(6 for AE, 5 for UTF8, 6 for data classification). I guess it's fine if we're consistent, but I'm just saying that other than the fedAuth function, the others don't really make a lot of sense.